src/Microsoft.Data.Analysis/IDataView.Extension.cs

```diff
@@ -0,0 +1,125 @@
+            if (selectColumns != null && selectColumns.Length > 0)
+            {
+                selectColumnsSet = new HashSet<string>(selectColumns);
+            }
+
+            List<DataViewSchema.Column> activeColumns = new List<DataViewSchema.Column>();
+            foreach (DataViewSchema.Column column in schema)
+            {
+                if (column.IsHidden || (selectColumnsSet != null && !selectColumnsSet.Contains(column.Name)))
+                {
+                    continue;
+                }
+
+                activeColumns.Add(column);
+                DataViewType type = column.Type;
+                if (type == BooleanDataViewType.Instance)
+                {
+                    columns.Add(new BooleanDataFrameColumn(column.Name));
+                }
+                else if (type == NumberDataViewType.Byte)
+                {
+                    columns.Add(new ByteDataFrameColumn(column.Name));
+                }
+                else if (type == NumberDataViewType.Double)
+                {
+                    columns.Add(new DoubleDataFrameColumn(column.Name));
+                }
+                else if (type == NumberDataViewType.Single)
+                {
+                    columns.Add(new SingleDataFrameColumn(column.Name));
+                }
+                else if (type == NumberDataViewType.Int32)
+                {
+                    columns.Add(new Int32DataFrameColumn(column.Name));
+                }
+                else if (type == NumberDataViewType.Int64)
+                {
+                    columns.Add(new Int64DataFrameColumn(column.Name));
+                }
+                else if (type == NumberDataViewType.SByte)
+                {
+                    columns.Add(new SByteDataFrameColumn(column.Name));
+                }
+                else if (type == NumberDataViewType.Int16)
+                {
+                    columns.Add(new Int16DataFrameColumn(column.Name));
+                }
+                else if (type == NumberDataViewType.UInt32)
+                {
+                    columns.Add(new UInt32DataFrameColumn(column.Name));
+                }
+                else if (type == NumberDataViewType.UInt64)
+                {
+                    columns.Add(new UInt64DataFrameColumn(column.Name));
+                }
+                else if (type == NumberDataViewType.UInt16)
+                {
+                    columns.Add(new UInt16DataFrameColumn(column.Name));
+                }
+                else if (type == TextDataViewType.Instance)
+                {
+                    columns.Add(new StringDataFrameColumn(column.Name));
+                }
+                else
+                {
+                    throw new NotSupportedException(String.Format(Microsoft.Data.Strings.NotSupportedColumnType, type.RawType.Name));
+                }
+            }
+
+            List<Delegate> activeColumnDelegates = new List<Delegate>();
```

Comment:
(minor) you know how many columns we are going to need by `activeColumns.Count`. You should be able to initialize the size here, or even better just create an array - no need for a List.